lib/IDE/CompletionInstance.cpp

```diff
@@ -324,14 +321,20 @@ bool CompletionInstance::performCachedOperaitonIfPossible(
     CompilerInstance::addAdditionalInitialImportsTo(newSF, implicitImport);
     newSF->enableInterfaceHash();
 
+    // Tell the compiler instance we've replaced the code completion file.
+    CI.setCodeCompletionFile(newSF);
+
     // Re-parse the whole file. Still re-use imported modules.
-    (void)oldState.takeCodeCompletionDelayedDeclState();
-    parseIntoSourceFile(*newSF, newBufferID, &oldState);
+    parseIntoSourceFile(*newSF, newBufferID);
     performNameBinding(*newSF);
     bindExtensions(*newSF);
 
-    assert(oldState.hasCodeCompletionDelayedDeclState() &&
-           oldState.getCodeCompletionDelayedDeclState().Kind == newInfo.Kind);
+    auto *reparsedState = newSF->getCodeCompletionParserState();
+    assert(reparsedState->hasCodeCompletionDelayedDeclState() &&
+           "Didn't find completion token?");
+
+    auto &reparsedInfo = reparsedState->getCodeCompletionDelayedDeclState();
+    assert(reparsedInfo.Kind == newInfo.Kind);
```

Comment:
`#ifndef NDEBUG ... #endif` to clarify this is not needed in non-assertion build. Also, could you `const`ify related methods?

Reply:
@rintaro Sure thing, how does this look?

Thanks for the review!

Reply:
Looks good. Thank you!